pgxn/neon/libpagestore.c

```diff
@@ -30,32 +30,31 @@
 #include "walproposer.h"
 #include "walproposer_utils.h"
 
-
 #define PageStoreTrace DEBUG5
 
 #define NEON_TAG "[NEON_SMGR] "
-#define neon_log(tag, fmt, ...) ereport(tag, \
-		(errmsg(NEON_TAG fmt, ## __VA_ARGS__), \
-		 errhidestmt(true), errhidecontext(true)))
+#define neon_log(tag, fmt, ...) ereport(tag,                                  \
```

Comment:
This PR screws up the formatting in the .c files. You can use `vendor/postgres-v14/src/tools/pgindent/pgindent` to reformat this according to the Postgres formatting guidelines. Or just revert all the formattting changes in C files in this PR. (I'm not sure when we last ran pgindent, so the files might not fully conform currently)

Reply:
I've rebased this PR on top of a [new commit](https://github.com/neondatabase/neon/pull/2433/commits/36fef0d30f79e0b9f9ad9b10948c1d598517be9d), that reformats the PG code based on the instruction from https://github.com/neondatabase/postgres/blob/114676d2edd5307226d9448ec467821fdb77467d/src/tools/pgindent/README

I've also added the `typedefs.list` file from the same commit, that I've used to reformat the C code.
So now the code is screwed up officially.